.github/workflows/soak.yml

```diff
@@ -0,0 +1,244 @@
+
+  build-comparison-image:
+    name: Build comparison 'soak-vector' container
+    runs-on: [self-hosted, linux, x64, general]
+    steps:
+      - uses: actions/checkout@v2.3.5
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          flavor: |
+            latest=false
+            prefix=
+            suffix=
+          images: ghcr.io/${{ github.repository }}/soak-vector
+          tags: type=raw, value=${{ github.sha }}
+
+      - name: Build and push 'soak-vector' image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          file: soaks/Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha, scope=${{ github.workflow }}
+          cache-to: type=gha, scope=${{ github.workflow }}
+          build-args: |
+            VECTOR_FEATURES=api,api-client,disk-buffer,enrichment-tables,rdkafka-plain,sinks,sources,sources-dnstap,transforms,unix,vendor-all,vrl-cli,datadog-pipelines
+
+  soak-baseline:
+    name: Soak (${{ matrix.target }}) - baseline
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak]
+    needs: [build-baseline-image]
+    strategy:
+      matrix:
+        target:
+          - datadog_agent_remap_datadog_logs
+          - syslog_humio_logs
+          - syslog_log2metric_humio_metrics
+          - syslog_loki
+          - syslog_regex_logs2metric_ddmetrics
+          - syslog_splunk_hec_logs
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+
+      - name: Download baseline-soak artifact
+        uses: actions/download-artifact@v1
+        with:
+          name: baseline-soak
+
+      - name: Run baseline experiment
+        run: |
+          mkdir -p /tmp/${{ matrix.target }}/
+          ./soaks/bin/soak_one.sh "false" ${{ matrix.target }} "baseline" `cat baseline-soak/sha` /tmp/${{ matrix.target }}
+
+      - name: Upload timing captures
+        uses: actions/upload-artifact@v1
+        with:
+          name: ${{ matrix.target}}-captures
+          path: /tmp/${{ matrix.target }}
+
+  soak-comparison:
+    name: Soak (${{ matrix.target }}) - comparison
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    runs-on: [self-hosted, linux, x64, soak]
+    needs: [build-comparison-image]
```

Comment:
This needs `soak-baseline` too, no?

This is related to another thought I had. Would it be possible to run all of the soaks (baseline and new) in parallel first and then compare the output rather than have the `new` soaks require the baseline? This could allow the workflow to run more quickly if we had sufficient runners to parallelize them all.

Reply:
It doesn't; what we're doing here is running exactly as you suggest. The dependency tree looks like:

```
build-baseline-image -> soak-baseline -------> analyze-results 
                                                                           /\
build-comparison-image -> soak-comparison ---|
```

Reply:
Perhaps I'm misunderstanding what you're suggesting?

Reply:
Yeah, `analyze-results` is where the two graphs converge, which is what allows the baseline and comparison jobs to run in parallel.